restic/cli/restore.go

```diff
@@ -221,6 +227,47 @@ func (r *Restic) linkRestorePaths(snapshot dto.Snapshot, restoreDir string) (str
 	return restoreRoot, nil
 }
 
+func (r *Restic) isRestoreSingleFile(log logr.Logger, snapshot dto.Snapshot) (bool, error) {
+	buf := bytes.Buffer{}
+
+	opts := CommandOptions{
+		Path:   r.resticPath,
+		Args:   r.globalFlags.ApplyToCommand("ls", "--json", snapshot.ID),
+		StdOut: &buf,
+	}
+
+	cmd := NewCommand(r.ctx, log, opts)
+	cmd.Run()
+	capturedStdOut := buf.String()
+
+	stdOutLines := strings.Split(capturedStdOut, "\n")
+
+	count := 0
+	for _, fileJSON := range stdOutLines {
+		node := &fileNode{}
+		err := json.Unmarshal([]byte(fileJSON), node)
+		if err != nil {
+			continue
```

Comment:
I'd prefer to log the error and not just discard it here. If for some reason the JSON is garbled it would probably take a long time to figure that out.

Reply:
Yeah, you are right, it's not good to skip error. I've just wrote this part of code like `extractFileNodes` function

I change this part of code and handling error to seem more understanding

Reply:
I fix it